src/storage/accessors/NormalizedDataAccessor.ts

```diff
@@ -0,0 +1,44 @@
+import type { Readable } from 'stream';
+import type { Patch } from '../../ldp/http/Patch';
+import type { Representation } from '../../ldp/representation/Representation';
+import type { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import { ensureTrailingSlash, trimTrailingSlashes } from '../../util/Util';
+import type { DataAccessor } from './DataAccessor';
+
+/**
+ * Abstract class that implements `getNormalizedMetadata` by calling the `getMetadata` with the incoming identifier
+ * and trying again by changing the trailing slash and trying again.
+ */
+export abstract class NormalizedDataAccessor implements DataAccessor {
```

Comment:
But I would not encode this functionality as a class. Can't `FileDataAccessor` (or perhaps better, `DataAccessorBasedStore`, possibly with a flag) have that functionality?